### PR TITLE
chore: Updated legacy documentation for CustomizedSystemMessages

### PR DESCRIPTION
**Pull request by mcollovati**
`CustomizedSystemMessages` javadocs had references to legacy `Application` class and behaviors.

Related to #15476

### CODE REVIEW

File: flow-server/src/main/java/com/vaadin/flow/server/CustomizedSystemMessages.java [hunk 1]
```diff
@@ -64,6 +68,8 @@ public void setSessionExpiredURL(String sessionExpiredURL) {
      * user. If the notification is disabled the user will be immediately
      * redirected to the URL returned by {@link #getSessionExpiredURL()}.
      *
+     * By default, the "session expired" notification" is disabled.
```
Comment (caalador) [inline flow-server/src/main/java/com/vaadin/flow/server/CustomizedSystemMessages.java:71]:
```suggestion
     * By default, the "session expired" notification is disabled.
```

File: flow-server/src/main/java/com/vaadin/flow/server/CustomizedSystemMessages.java [hunk 2]
```diff
@@ -119,6 +127,9 @@ public void setInternalErrorURL(String internalErrorURL) {
      * user. If the notification is disabled the user will be immediately
      * redirected to the URL returned by {@link #getInternalErrorURL()}.
      *
+     * By default, the "internal error" notification" is enabled.
```
Comment (caalador) [inline flow-server/src/main/java/com/vaadin/flow/server/CustomizedSystemMessages.java:130]:
```suggestion
     * By default, the "internal error" notification is enabled.
```

File: flow-server/src/main/java/com/vaadin/flow/server/CustomizedSystemMessages.java [hunk 3]
```diff
@@ -164,6 +177,8 @@ public void setCookiesDisabledURL(String cookiesDisabledURL) {
      * user. If the notification is disabled the user will be immediately
      * redirected to the URL returned by {@link #getCookiesDisabledURL()}.
      *
+     * By default, the "cookies disabled" notification" is enabled.
```
Comment (caalador) [inline flow-server/src/main/java/com/vaadin/flow/server/CustomizedSystemMessages.java:180]:
```suggestion
     * By default, the "cookies disabled" notification is enabled.
```

**Review by caalador — DISMISSED** [review]

File: flow-server/src/main/java/com/vaadin/flow/server/CustomizedSystemMessages.java [hunk 4]
```diff
@@ -20,38 +20,42 @@
...
  * <p>
- * Vaadin gets the SystemMessages from your application by calling a static
- * getSystemMessages() method. By default the Application.getSystemMessages() is
- * used. You can customize this by defining a static
- * MyApplication.getSystemMessages() and returning CustomizedSystemMessages.
- * Note that getSystemMessages() is static - changing the system messages will
- * by default change the message for all users of the application.
+ * Vaadin gets the SystemMessages from the {@link SystemMessagesProvider}
+ * configured in {@link VaadinService}. You can customize this by creating a
+ * {@link VaadinServiceInitListener} that sets an instance on
```
Comment (tepi) [inline flow-server/src/main/java/com/vaadin/flow/server/CustomizedSystemMessages.java:25]: Should this be "an instance of" ?

Comment (mcollovati) [inline flow-server/src/main/java/com/vaadin/flow/server/CustomizedSystemMessages.java:25]:
```suggestion
 * {@link VaadinServiceInitListener} that sets an instance of
```

Comment (mcollovati) [reply flow-server/src/main/java/com/vaadin/flow/server/CustomizedSystemMessages.java]: Updated

**Review by tepi — APPROVED** [review]

**Pull request merged**